src/hooks/useEarnings.js

```diff
@@ -34,5 +35,13 @@ export default function useEarnings() {
     setEarnings([...earnings.slice(0, earnings.length - 1), newEarning])
   }
 
-  return { lastEarning, distribute, addEarning, earnings }
+  function spendCoins(type, value) {
+    const lastSpending = {
+      ...lastEarning,
```

Comment:
Spending should use the total of all coins not just the last earning, that's why it only works for the very first time.